integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -1088,3 +1100,66 @@ def do_it(spark):
         df = two_col_df(spark, StringGen('k{1,5}'), ArrayGen(MapGen(StringGen('a{1,5}', nullable=False), StringGen('[ab]{1,5}'))))
         return df.groupBy('a').agg(f.min(df.b[1]["a"]))
     assert_gpu_and_cpu_are_equal_collect(do_it)
+
+
+# Tests for standard deviation and variance aggregations.
+@approximate_float
+@ignore_order
+@incompat
+@pytest.mark.parametrize('data_gen', _init_list_with_nans_and_no_nans, ids=idfn)
+@pytest.mark.parametrize('conf', get_params(_confs, params_markers_for_confs), ids=idfn)
+def test_groupby_std_variance(data_gen, conf):
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark : gen_df(spark, data_gen, length=100),
+        "data_table",
+        'select ' +
+        'stddev(b),' +
+        'stddev_pop(b),' +
+        'stddev_samp(b),' +
+        'variance(b),' +
+        'var_pop(b),' +
+        'var_samp(b)' +
+        ' from data_table group by a',
+        conf=conf)
+
+# This test also includes the test above thus let remove the test above when this one can be enabled.
+@pytest.mark.xfail(reason='GPU-casting directly from decimal to double is not yet supported')
```

Comment:
please put a link to the issue here, like other xfails https://github.com/NVIDIA/spark-rapids/issues/3610